test-suite-kotlin/build.gradle

```diff
@@ -29,12 +29,7 @@ tasks.named('test') {
 
 compileTestKotlin {
     kotlinOptions {
-        jvmTarget = '1.8'
+        jvmTarget = '17'
         javaParameters = true
     }
 }
-
-java {
```

Comment:
should we set: 

```
java {
    sourceCompatibility = JavaVersion.toVersion("17")
    targetCompatibility = JavaVersion.toVersion("17")
}
```

Create a test-base in buildScript to avoid repetition.

Reply:
I have created a PR I have created a PR https://github.com/micronaut-projects/micronaut-email/pull/135